File: Ai algorithm implementation/Iterative_DFS.py
def dfs_recursive(adj, visited, node):
    if visited[node]:
        return
    visited[node] = True
    print(node, end=' ')
    
    for child in adj[node]:
        if not visited[child]:
            dfs_recursive(adj, visited, child)

adj = {
    1: [2, 3],
    2: [1, 4, 5],
    3: [1],
    4: [2],
    5: [2]
}

visited = [False] * 6
print("DFS (Recursive): ", end='')
dfs_recursive(adj, visited, 2)

File: Ai algorithm implementation/Iterative_DFS.cpp
#include <bits/stdc++.h>
using namespace std;

void dfs_recursive_style(vector<int> adj[], vector<bool> &visited, int node) {
    if (visited[node]) return;
    visited[node] = true;
    cout << node << " ";

    for (int child : adj[node]) {
        if (!visited[child])
            dfs_recursive_style(adj, visited, child);
    }
}

int main() {
    vector<int> adj[6];
    adj[1] = {2, 3};
    adj[2] = {1, 4, 5};
    adj[3] = {1};
    adj[4] = {2};
    adj[5] = {2};

    vector<bool> visited(6, false);
    cout << "DFS (Recursive Style): ";
    dfs_recursive_style(adj, visited, 2);
}
